src/Test.UnitTests.Sarif/Baseline2/WhatComparerTests.cs

```diff
@@ -69,101 +75,111 @@ public void WhatComparer_Basics()
 
             left.PartialFingerprints = null;
             right.PartialFingerprints = null;
 
             // Fallback
             // ========
 
-            // Don't match via fallback (Message doesn't match)
-            Assert.False(WhatComparer.MatchesWhat(eLeft, eRight));
+            // Don't match via fallback (Message doesn't match).
+            WhatComparer.MatchesWhat(eLeft, eRight).Should().BeFalse();
 
-            // Verify match if messages match
+            // Verify match if messages match.
             right.Message.Text = "One";
-            Assert.True(WhatComparer.MatchesWhat(eLeft, eRight));
+            WhatComparer.MatchesWhat(eLeft, eRight).Should().BeTrue();
 
-            // Verify no match if not enough matching Properties
+            // Verify no match if not enough matching Properties.
 
-            // Don't match with if only Property is a non-match
+            // Don't match if the only Property is a non-match.
             left.SetProperty("First", "0001");
             right.SetProperty("First", "1001");
-            Assert.False(WhatComparer.MatchesWhat(eLeft, eRight));
+            WhatComparer.MatchesWhat(eLeft, eRight).Should().BeFalse();
 
-            // Match if only one matches
+            // Match if only one matches.
             right.SetProperty("First", "0001");
-            Assert.True(WhatComparer.MatchesWhat(eLeft, eRight));
+            WhatComparer.MatchesWhat(eLeft, eRight).Should().BeTrue();
 
-            // Match if one of two (50%) matches
+            // Match if one of two (50%) matches.
             left.SetProperty("Second", "0002");
             right.SetProperty("Second", "1002");
-            Assert.True(WhatComparer.MatchesWhat(eLeft, eRight));
+            WhatComparer.MatchesWhat(eLeft, eRight).Should().BeTrue();
 
-            // Don't match if one of three (under 50%) match
+            // Don't match if one of three (under 50%) match.
             left.SetProperty("Third", "0003");
             right.SetProperty("Third", "1003");
-            Assert.False(WhatComparer.MatchesWhat(eLeft, eRight));
+            WhatComparer.MatchesWhat(eLeft, eRight).Should().BeFalse();
 
-            // Match for two of four matches
+            // Match for two of four matches.
             left.SetProperty("Fourth", "0004");
             right.SetProperty("Fourth", "0004");
-            Assert.True(WhatComparer.MatchesWhat(eLeft, eRight));
+            WhatComparer.MatchesWhat(eLeft, eRight).Should().BeTrue();
 
-            // Don't match when names are different
+            // Don't match when names are different.
             left.SetProperty("Fifth_Left", "0005");
             right.SetProperty("Fifth_Right", "0005");
-            Assert.False(WhatComparer.MatchesWhat(eLeft, eRight));
+            WhatComparer.MatchesWhat(eLeft, eRight).Should().BeFalse();
+
+            // Match when non-string properties are equal.
```

Comment:
>// Match when non-string properties are equal. [](start = 12, length = 46)

Test the new ability to compare non-string-valued properties. Before I added the new APIs to `PropertyBagHolder`, and modified `WhatComparer` to use them, these two tests would have crashed.